modules/op-about-stepactions.adoc

```diff
@@ -0,0 +1,81 @@
+// This module is included in the following assemblies:
+// * about/understanding-openshift-pipelines.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="about-stepactions_{context}"]
+= StepAction
+
+A step is a part of a task. If you define a step in a task, you cannot reference this step from another task.
+
+However, optionally, you can define a `StepAction` custom resource (CR). This CR contains the action that a step performs. You can reference a `StepAction` object from a step to create a step that performs the action. You can also use resolvers to reference a `StepAction` object that is available in another namespace or from an external source.
+
+The following examples shows a `StepAction` CR named `apply-manifests-action`. This action applies manifests from a source tree to your {OCP} environment.
+
+[source,yaml]
+----
+apiVersion: tekton.dev/v1
+kind: StepAction
+metadata:
+  name: apply-manifests-action
+spec:
+  params:
+  - name: working_dir
+    description: The working directory where the source is located
+    type: string
+    default: "/workspace/source"
+  - name: manifest_dir
+    description: The directory in source that contains yaml manifests
+    type: string
+    default: "k8s"
+  results:
+  - name: output
+    description: The output of the oc apply command
+  image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+  env:
+  - name: WORKING_DIR
+    value: $(params.working_dir)
+  - name: MANIFEST_DIR
+    value: $(params.manifest_dir)
+  script: |
+      #!/usr/bin/env bash
+      cd $WORKING_DIR
+      oc apply -f "$MANIFEST_DIR" | tee $(results.output)
+----
+
+The `StepAction` CR does not include definitions of workspaces. Instead, the action expects that the task that includes the action also provides the mounted source tree, typically using a workspace.
+
+A `StepAction` object can define parameters and results. When you reference this object, you must specify the values for the parameters of the `StepAction` object in the definition of the step. The results of the `StepAction` object automatically become the results of the step.
+
+[IMPORTANT]
+====
+In order to avoid malicious attacks that use the shell, `StepAction` CR does not support using parameter values in a `script` value. Instead, you must use the `env:` section to define environment variables that contain the parameter values.
+====
+
+The following example task includes a step that references the `apply-manifests-action`  object, provides the necessary parameters, and uses the result.
+
+[source,yaml]
+----
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: apply-manifests-with-action
+spec:
+  workspaces:
+  - name: source
+  params:
+  - name: manifest_dir
+    description: The directory in source that contains yaml manifests
+    type: string
```

Comment:
string is default value, this line can be omitted

Reply:
Maybe I want to keep it in to demonstrate that one can set a type - unless we can invent an example that includes another parameter type. Or atre other types really marginal cases, do strings as parameters normally suffice?